graylog2-server/src/main/java/org/graylog2/users/UserRolesHelper.java

```diff
@@ -0,0 +1,72 @@
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.graylog2.users;
+
+import com.google.common.base.Predicates;
+import com.google.common.collect.Collections2;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Iterables;
+import org.graylog2.database.NotFoundException;
+import org.graylog2.plugin.database.users.User;
+import org.graylog2.security.InMemoryRolePermissionResolver;
+import org.graylog2.shared.users.Role;
+import org.graylog2.shared.users.Roles;
+
+import javax.inject.Inject;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class UserRolesHelper {
+    private final RoleService roleService;
+    private final InMemoryRolePermissionResolver inMemoryRolePermissionResolver;
+
+    @Inject
+    public UserRolesHelper(RoleService roleService, InMemoryRolePermissionResolver inMemoryRolePermissionResolver) {
+        this.roleService = roleService;
+        this.inMemoryRolePermissionResolver = inMemoryRolePermissionResolver;
+    }
+
+    public Set<String> getRoleNames(User user) throws NotFoundException {
```

Comment:
This method should really be placed either in `RoleService` or in `UserService` (depending on the point of view, I'd place it in `UserService`).